tests/FileRemovalTest.php

```diff
@@ -0,0 +1,131 @@
+<?php
+
+/*
+ * This file is part of PHP CS Fixer.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *     Dariusz Rumiński <dariusz.ruminski@gmail.com>
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace PhpCsFixer\Tests;
+
+use org\bovigo\vfs\vfsStream;
+use PhpCsFixer\FileRemoval;
+
+/**
+ * @author ntzm
+ *
+ * @internal
+ *
+ * @covers \PhpCsFixer\FileRemoval
+ */
+final class FileRemovalTest extends TestCase
+{
+    private static $removeFilesOnTearDown = true;
```

Comment:
just by reading the code, one may thing that as soon as `testShutdownRemovesObservedFilesSetup` is executed then this flag is set to false, thus we would never cleanup those files.
IMO it's worth to document the background between this variable and correlation with `testShutdownRemovesObservedFilesSetup`.

Solution works like a charm, but is super counter intuitive, at least for me.

Reply:
Yeah I'm not 100% happy with it, but I can't figure out a better way of doing it, it feels like we're abusing the `@runInSeparateProcess` a little. Will document 👍

Reply:
exactly. and especially for that, docu is nice to have ;)